src/mapproxyUtils/mapproxySeed.ts

```diff
@@ -0,0 +1,264 @@
+import { promises as fsp } from 'node:fs';
+import { $ } from 'zx';
+import { dump } from 'js-yaml';
+import { Logger } from '@map-colonies/js-logger';
+import { inject, singleton } from 'tsyringe';
+import { SERVICES } from '../common/constants';
+import { IConfig, ISeedBase, ITaskSeedOptions } from '../common/interfaces';
+import { MapproxyConfigClient } from '../clients/mapproxyConfig';
+/* eslint-disable @typescript-eslint/naming-convention */
+import { BaseCache, Cleanup, Seed, seedsSchema, cleanupsSchema, baseSchema } from '../common/schemas/seeds';
+import { Coverage, coveragesSchema } from '../common/schemas/coverages';
+import { SeedMode } from '../common/enums';
+import { fileExists, isValidDateFormat, zoomComparison } from '../common/validations';
+
+$.verbose = false;
+
+@singleton()
+export class MapproxySeed {
+  private readonly mapproxyYamlDir: string;
+  private readonly seedYamlDir: string;
+  private readonly geometryCoverageFilePath: string;
+  private readonly seedConcurrency: number;
+  private readonly mapproxySeedProgressDir: string;
+
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    private readonly mapproxyConfigClient: MapproxyConfigClient
+  ) {
+    this.mapproxyYamlDir = this.config.get<string>('mapproxy.mapproxyYamlDir');
+    this.seedYamlDir = this.config.get<string>('mapproxy.seedYamlDir');
+    this.geometryCoverageFilePath = this.config.get<string>('mapproxy.geometryTxtFile');
+    this.seedConcurrency = 5;
+    this.mapproxySeedProgressDir = this.config.get<string>('mapproxy.seedProgressFileDir');
+  }
+
+  public async runSeed(task: ITaskSeedOptions): Promise<void> {
+    this.logger.info({
+      msg: `processing seed for job of ${task.layerId}`,
+      layerId: task.layerId,
+      grid: task.grid,
```

Comment:
add jobId and taskId to **all** logs in this file - pass them to methods if needed

Reply:
done